### ISSUE TITLE
`DEPRECATION: Legacy editable install` during `pip install -e .` step of `install_sct`

### ISSUE THREAD

**Issue by joshuacwnewton** [spinalcordtoolbox/spinalcordtoolbox#4599]
The full error message is:

```
Installing collected packages: spinalcordtoolbox
  DEPRECATION: Legacy editable install of spinalcordtoolbox==6.4.dev0 from file:///home/runner/sct_6.4.dev0 (setup.py develop) is deprecated. pip 25.0 will enforce this behaviour change. A possible replacement is to add a pyproject.toml or enable --use-pep517, and use setuptools >= 64. If the resulting installation is not behaving as expected, try using --config-settings editable_mode=compat. Please consult the setuptools documentation for more information. Discussion can be found at https://github.com/pypa/pip/issues/11457
  Running setup.py develop for spinalcordtoolbox
Successfully installed spinalcordtoolbox
```

This error has made me realize that, while we pin our _pip dependencies_ via `pip freeze`, we don't pin the dependencies that conda provides (e.g. `pip` itself), meaning that all past SCT versions may be susceptible to breakage once `pip==25.0` is released. (It's not clear to me if pip 25.0 will fail, or if it will automatically use the `--use-pep517` option.)

Anyway, testing `--use-pep517` locally shows no issue:

```
(venv_sct) jn/ads-install-script|joshua@monarch:~/repos/spinalcordtoolbox$ pip install -e . --use-pep517
Obtaining file:///home/joshua/repos/spinalcordtoolbox
  Installing build dependencies ... done
  Checking if build backend supports build_editable ... done
  Getting requirements to build editable ... done
  Preparing editable metadata (pyproject.toml) ... done
Building wheels for collected packages: spinalcordtoolbox
  Building editable for spinalcordtoolbox (pyproject.toml) ... done
  Created wheel for spinalcordtoolbox: filename=spinalcordtoolbox-6.4.dev0-0.editable-py3-none-any.whl size=9784 sha256=a395c601e67e1ff13fb3db19825d68c81503207613e6e967a4fcbb20ad0e87b8
  Stored in directory: /tmp/pip-ephem-wheel-cache-k5304kt1/wheels/12/d4/75/b75a0b0fa690f5d1477b7d2b91f862af1d4ece58ca4c5ba5a5
Successfully built spinalcordtoolbox
Installing collected packages: spinalcordtoolbox
  Attempting uninstall: spinalcordtoolbox
    Found existing installation: spinalcordtoolbox 6.4.dev0
    Uninstalling spinalcordtoolbox-6.4.dev0:
      Successfully uninstalled spinalcordtoolbox-6.4.dev0
Successfully installed spinalcordtoolbox-6.4.dev0
```

So, it should at the very least be an easy fix.

**Comment by mguaypaq** [comment]
I'm confused: according to [this page](https://pip.pypa.io/en/stable/reference/build-system/#controlling-which-build-system-interface-is-used):
> The [`--use-pep517`](https://pip.pypa.io/en/stable/cli/pip_install/#install-use-pep517) flag (and corresponding environment variable: `PIP_USE_PEP517`) can be used to force all packages to build using the `pyproject.toml` based build system interface.

But we don't have a `pyproject.toml` file, so how does it work?

**Comment by joshuacwnewton** [comment]
My current understanding is that:

- The actual [PEP517](https://peps.python.org/pep-0517/) is kind of a red herring. It provides the nuts and bolts specification, but the actually relevant PEP is [PEP660](https://peps.python.org/pep-0660/), which builds on top of PEP517.
- "Editable installations" is a concept that started out being inherent to `setup.py`/`setuptools`. 
- When `pyproject.toml` was introduced, it didn't initially support editable installations. PEP660 changed this by adding a new spec that defines what tempfiles are necessary in order to generate an "editable wheel". The spec itself isn't specific to `pyproject.toml`, but `pyproject.toml` is what necessitated the spec in the first place.
- From what I understand, `--use-pep517` tells `pip` to take the metadata from `setup.py` (e.g. `console_scripts`, etc.) and convert it into the PEP660 format. 
   - Case in point, I set a breakpoint at the "  Preparing editable metadata (pyproject.toml) ... done" part of the installation, and found that the function that spawns this message looks like this:
      ```python
      def generate_editable_metadata(
          build_env: BuildEnvironment, backend: BuildBackendHookCaller, details: str
      ) -> str:
          """Generate metadata using mechanisms described in PEP 660.
      
          Returns the generated metadata directory.
          """
      ```
- In other words, this function isn't necessarily specific to `pyproject.toml`. And, when I check which files are generated in the tmpdir, it's just the `.egg-info` and `.dist-info`:
  ![image](https://github.com/user-attachments/assets/4dbf469f-aeba-404c-94ae-fe2e25c8d620)
  So I presume `pip` is just taking `setup.py` and masquerading its metadata as if it were defined by `pyproject.toml`?
